openedx/features/assessment/tests/test_helpers.py

```diff
@@ -0,0 +1,186 @@
+        assert mock_reset_score.called_with(anonymous_user.anonymous_user_id,
+                                            self.source_course.id.to_deprecated_string(),
+                                            unicode(self.block_usage_locator))
+        assert mock_log_info.called_once
+
+    @mock.patch('openedx.features.assessment.helpers.log.info')
+    @mock.patch('openedx.features.assessment.helpers.log.warn')
+    @mock.patch('openedx.features.assessment.helpers.reset_score')
+    @mock.patch('openedx.features.assessment.helpers.set_score')
+    def test_autoscore_ora_with_no_submission(self, mock_set_score, mock_reset_score, mock_log_war, mock_log_info):
+        """ Test auto score without any ora submissions """
+
+        anonymous_user = AnonymousUserIdFactory(user=self.user, course_id=self.source_course.id)
+
+        student = {
+            'anonymous_user_id': anonymous_user.anonymous_user_id
+        }
+        helpers.autoscore_ora(self.source_course.id, unicode(self.block_usage_locator), student)
+
+        self.assertEqual(Assessment.objects.all().count(), 0)
+        self.assertEqual(AssessmentPart.objects.all().count(), 0)
+        assert mock_set_score.not_called
+        assert mock_reset_score.not_called
+        assert mock_log_war.called_once
+        assert mock_log_info.not_called
+
+    @data(
+        (TWO_POINT_RUBRIC_DICTIONARY, {'Ideas': 'Fair', 'Content': 'Fair'}, 4, 4),
+        (THREE_POINT_RUBRIC_DICTIONARY, {'Ideas': 'Fair', 'Content': 'Fair'}, 4, 6)
+    )
+    @unpack
+    def test_select_options_with_two_point_rubric(self, rubric_dict, expected_options_selected,
+                                                  expected_earned_points, expected_possible_points):
+        """ Validate rubric points calculation """
+
+        actual_options_selected, actual_earned_points, actual_possible_points = helpers.select_options(rubric_dict)
+
+        self.assertEqual(actual_options_selected, expected_options_selected)
+        self.assertEqual(actual_earned_points, expected_earned_points)
+        self.assertEqual(actual_possible_points, expected_possible_points)
+
+    def test_get_philu_bot(self):
+        """ Verify that bot user is created if not already exists """
+        helpers.get_philu_bot()
+        self.assertIs(User.objects.filter(username=PHILU_BOT_NAME).exists(), True)
+
+    @mock.patch('openedx.features.assessment.helpers.modulestore')
+    def test_get_rubric_for_course(self, mock_modulestore):
+        """ Test rubric for course """
+        mock_get_item = mock_modulestore().get_item()
+        mock_get_item.prompts.return_value = 'mock prompts'
+        mock_get_item.rubric_criteria.return_value = 'mock criteria'
+        expected_rubric_dict = {
+            'prompts': mock_get_item.prompts,
+            'criteria': mock_get_item.rubric_criteria
+        }
+        # Find the associated rubric for that course_id & item_id
```

Comment:
One space here would be nice. It will split code in three parts test data, call to function, assert